### PR TITLE
fix: fix ingress annotations format

### PR DESCRIPTION
**Pull request by ricolin**
fix https://github.com/vexxhost/atmosphere/issues/1278

### CODE REVIEW

File: roles/ingress/defaults/main.yml [hunk 1]
```diff
@@ -38,5 +38,9 @@ ingress_class_name: "{{ atmosphere_ingress_class_name }}"
 #               annotations to all Ingress resources.
 ingress_annotations: {}
 
+# Seperate cert manager annotations so ingress_annotations can be override.
+ingress_cert_annotations:
+  cert-manager.io/common-name: "{{ ingress_host }}"
```
Comment (mnaser) [inline roles/ingress/defaults/main.yml:43]: @ricolin is there a reason we can't keep the existing template and not introduce a new variable to handle this?

**Review by mnaser — CHANGES_REQUESTED** [review]

File: roles/ingress/defaults/main.yml (hunk 1 above)
Comment (ricolin) [reply roles/ingress/defaults/main.yml:43]: depends on how we like the override to works here.
We can totally write a hard code dict in template like ansible.builtin.combine({cert-manager.io/common-name: "{{ ingress_host }}"}
or put this cert-manager stuff into `ingress_annotations`

Comment (ricolin) [reply roles/ingress/defaults/main.yml:43]: like this?
```
  annotations:
      {{ { cert-manager.io/common-name: {{ ingress_host }} } | combine(ingress_annotations, recursive=True) | combine(ingress_global_annotations, recursive=True) | to_nice_yaml | indent(4) }
```

Comment (ricolin) [reply roles/ingress/defaults/main.yml]: updated, please check :)

Comment (mnaser) [reply roles/ingress/defaults/main.yml]: Is there no way of keeping the annotation hard coded the way it is and then adding the extra annotation on top of it?

Comment (ricolin) [reply roles/ingress/defaults/main.yml]: Another way is to check keys before parse to nice yaml function so {} won't be generated.
Just updated again. Please check. Thanks

**Comment by ricolin** [comment]
recheck

**Comment by ricolin** [comment]
recheck

**Comment by ricolin** [comment]
recheck

**Review by mnaser — APPROVED** [review]

**Pull request merged**